src/mscorlib/shared/System/StringComparer.cs

```diff
@@ -276,5 +276,11 @@ public override int GetHashCode(string obj)
         // Equals/GetHashCode methods for the comparer itself. 
         public override bool Equals(object obj) => obj is OrdinalIgnoreCaseComparer;
         public override int GetHashCode() => nameof(OrdinalIgnoreCaseComparer).GetHashCode();
+
+        public void GetObjectData(SerializationInfo info, StreamingContext context)
+        {
+            info.SetType(typeof(OrdinalComparer));
+            info.AddValue("_ignoreCase", true);
```

Comment:
@jkotas This will work when we serialize from Core to Desktop but not the other way around because OrdinalComparer in Core doesn't use the _ignoreCase value. We need some routine to deserialize to OrdinalIgnoreCaseComparer instead of OrdinalComparer if _ignoreCase is true.

I found some info about creating a custom SerializationBinder but this is a) a hassle and b) won't work because we don't have the _ignoreCase value at that time to decide to which type to deserialize.